### ISSUE TITLE
Shell stops responding with tmux is executed with default

### ISSUE THREAD

**Issue by KunalSin9h** [warpdotdev/Warp#4216]
### Discord username (optional)

kunalsin9h

### Describe the bug

Os: Arch Linux
Installed: `.pkg.tar.zst`

I have a small code in my `.zshrc` with run tmux all the time by default, like when i open new terminal `tmux` will also be initialized, and i will be in tmux. 

This is causing `warp` to stop working (only shell) buttons are working but no keyboard interaction is happening.

When i remove the code what run `tmux` everything works and manually running `tmux` also works.

### To reproduce

1. Use bellow code to automatically start `tmux` on every new `zsh` session.
```bash
if command -v tmux &> /dev/null && [ -n "$PS1" ] && [[ ! "$TERM" =~ screen ]] && [[ ! "$TERM" =~ tmux ]] && [ -z "$TMUX" ]; then
  exec tmux
fi
```
2. Open `warp`

### Operating system

Linux

### Operating system and version

Arch Linux

### Shell Version

zsh 5.9

### Current Warp version

Newest warp for linux v0.2024.02.20.08.02.stable_01-1-x86_64

### Regression

No, this bug or issue has existed throughout my experience using Warp

### Does this block you from using Warp daily?

No

### Is this a Warp specific issue? (i.e. does it happen in Terminal, iTerm, Kitty, etc.)

Yes, this I confirmed this only happens in Warp, not other terminals.

### Warp Internal (ignore): linear-label:b9d78064-c89e-4973-b153-5178a31ee54e

None

**Comment by siraht** [comment]
This is also happening for me using the .deb on Ubuntu 22.04, with bash and .bashrc

**Comment by bitspittle** [comment]
Just installed Warp for linux and ran into this. When I started Warp, my shell worked, but Warp showed a dismissible popup saying something like "it seems like it's taking a long time for my shell to start up...."

But I guessed it was tmux and updated my login script to stop running it on startup. Warp's functionality from my testing so far should be all I need to replace it completely.

Tmux sets the env var TMUX if it is running, so Warp can check for that if it wants to share a more informative error message / guidance in this case.

**Comment by KunalSin9h** [comment]
@siraht @bitspittle i think using tmux in warp does not make any sense, since when we execute tmux in warp, we end up in normal shell, at all the features of warp are useless.

**Comment by bitspittle** [comment]
Correct. I disabled tmux. However, it was a lucky guess at the time. It was not obvious that tmux was the culprit. It would have been nice if Warp detected this case and informed me about what was going on, maybe sending me to a docs page about how to fix it (for example)

**Comment by KunalSin9h** [comment]
@bitspittle or if detect tmux, then just ignore and open normally but when i open my xterm or urxvt then i can use my tmux default spawn.
